### ISSUE TITLE
Detect open() calls without with operator

### ISSUE THREAD

**Issue by EugeneZelenko** [PyCQA/pylint#3413]
It'll be great to have check that will catch call to open() made without with operator. Probably check should work for other resource-allocating library calls.

**Comment by DudeNr33** [comment]
I'd be willing to work on this.
From what I've seen this would be a good addition to the existing ``RefactoringChecker`` in the form of a ``consider-using-with`` refactoring message.

@Pierre-Sassoulas let me know what you think.

**Comment by Pierre-Sassoulas** [comment]
Yes, nice refactoring message name :) ! I assigned you to the issue, thank you !

**Comment by benburk** [comment]
Thanks for the great addition! Unfortunately I'm not seeing it trigger on my end.

```bash
$ cat tmp.py
open('test.txt')
$ pylint tmp.py
************* Module tmp
tmp.py:1:0: C0304: Final newline missing (missing-final-newline)
tmp.py:1:0: C0114: Missing module docstring (missing-module-docstring)

----------------------------------------------------------------------
Your code has been rated at -10.00/10 (previous run: -10.00/10, +0.00)

$ pylint --version
pylint 2.8.2
astroid 2.5.6
Python 3.9.4 (default, Apr  9 2021, 09:32:38) 
[Clang 10.0.0 ]```

**Comment by DudeNr33** [comment]
That is because you did not assign the return value to anything. 
Is this a legitimate use case for you? 
I guess just the open call itself not, but you could do something like ``open("foo.txt", "w").write("hello world")``. And in this case it would currently not trigger as well.

**Comment by EugeneZelenko** [comment]
But isn't file will not be closed in case of `open("foo.txt", "w").write("hello world")`'? So it'll be reasonable to extend check to cover such cases too.

**Comment by DudeNr33** [comment]
Yes, it would remain open.
I agree it would make sense, it just wasn't a use case I had in mind when implementing the check.

**Comment by hippo91** [comment]
@DudeNr33 should i reopen this issue?

**Comment by DudeNr33** [comment]
@hippo91 as this issue is already referenced as "closed" in previous release notes, I'd say no.
I will see what needs to be changed to detect the use case of ``open()`` outside an assignment, and if you prefer to have an issue to link the pull request against I can open a new one. Otherwise I'd just submit the PR.
